[mlir][complex] Add a numerically-stable lowering for complex.expm1.

pifon2a opened the pull request:
The current conversion to Standard in the MLIR repo is not stable for small imag(arg).

akuegel commented on mlir/lib/Conversion/ComplexToStandard/ComplexToStandard.cpp:
```diff
@@ -520,29 +520,94 @@ struct ExpOpConversion : public OpConversionPattern<complex::ExpOp> {
+    Value realResult = b.create<arith::AddFOp>(
+        b.create<arith::MulFOp>(expm1Real, cosImag, fmf), cosm1Imag, fmf);
+
+    Value imageIsZero = b.create<arith::CmpFOp>(arith::CmpFPredicate::OEQ, imag,
```
nit: imageIsZero -> imagIsZero?

pifon2a replied:
done

pifon2a committed:
[mlir][complex] Add a numerically-stable lowering for complex.expm1.

The current conversion to Standard in the MLIR repo is not stable for small
imag(arg).
